.github/workflows/ci.yml

```
@@ -0,0 +1,198 @@
+        go tool cover -func=coverage.out | tail -1 | awk '{print "Total coverage: " $3}'
+        COVERAGE=$(go tool cover -func=coverage.out | tail -1 | awk '{print $3}' | sed 's/%//')
+        echo "Coverage: ${COVERAGE}%"
+        if (( $(echo "$COVERAGE < 70.0" | bc -l) )); then
+          echo "Coverage ${COVERAGE}% is below minimum threshold of 70%"
+          exit 1
+        fi
+      shell: bash
+
+    - name: Upload coverage to Codecov
+      if: matrix.os == 'ubuntu-latest' && matrix.go-version == '1.24.5'
+      uses: codecov/codecov-action@v3
```

actionlint (1.7.7)

the runner of "codecov/codecov-action@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)